src/services/kratos/auth-device-account.ts

```diff
@@ -0,0 +1,79 @@
+import * as jose from "node-jose"
+
+import { getAppCheckConfig, getJwksArgs, isCI, isDev } from "@config"
+
+import jwksRsa from "jwks-rsa"
+import jsonwebtoken from "jsonwebtoken"
+
+import { KratosError } from "./errors"
+
+import { AuthWithPhonePasswordlessService } from "./auth-phone-no-password"
+
+export const AuthWithDeviceAccountService = () => {
+  const upgradeToPhoneSchema = async ({
+    phone,
+    deviceId,
+  }: {
+    phone: PhoneNumber
+    deviceId: UserId
+  }): Promise<SessionToken | KratosError> => {
+    // 1. create kratos account
+    // 2. kratos webhook calls /kratos/registration to update mongo
+    //    account/user collection to ref kratos uuid instead of device id
+    const authService = AuthWithPhonePasswordlessService()
+    const kratosResult = await authService.createIdentityWithDeviceId({ phone, deviceId })
+    if (kratosResult instanceof Error) return kratosResult
+    return kratosResult.sessionToken
+  }
+
+  const verifyJwt = async (token: string) => {
+    const audience = getAppCheckConfig().audience
+    const issuer = getAppCheckConfig().issuer
+    const jwksUri = getAppCheckConfig().jwksUri
+
+    // Get the JSON web keys (JWKS)
+    let client = await jwksRsa({
+      jwksUri,
+    })
+    let jwkJson
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    jwkJson = (await client.getKeys()) as Array<any>
+
+    // Decode the token
+    const decodedToken = jsonwebtoken.decode(token, { complete: true })
+    if (!decodedToken) return Error("problem with jwt")
+
+    // Find the kid (Key Id) used to sign the token
+    const kid = decodedToken.header.kid
+    let kidJwk = jwkJson.find((j) => j.kid === kid)
+
+    // if in dev environment, use local jwks.json if you dont have access to firebase appcheck debug
+    if (!kidJwk && (isDev || isCI)) {
```

Comment:
This check shouldn't be made here at runtime every call. Just once at startup time and injected.

Reply:
refactored this code to use existing function sendOathkeeperRequest